src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -113,4 +115,408 @@ private async Task<HttpResponseMessage> GetDataSet(Guid dataSetId)
+
+            var expectedPageCount = dataSetVersions
+                .Skip((page - 1) * pageSize)
+                .Take(pageSize)
+                .Count();
+
+            Assert.NotNull(content);
+            Assert.Equal(page, content.Paging.Page);
+            Assert.Equal(pageSize, content.Paging.PageSize);
+            Assert.Equal(numberOfAvailableDataSetVersions, content.Paging.TotalResults);
+            Assert.Equal(expectedPageCount, content.Results.Count);
+        }
+
+        [Fact]
+        public async Task MultipleDataSetVersionsAvailableForRequestedDataSet_Returns200_OrderedCorrectly()
+        {
+            DataSet dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusPublished();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            var dataSetVersions = DataFixture
+                .DefaultDataSetVersion(
+                    filters: 1,
+                    indicators: 1,
+                    locations: 1,
+                    timePeriods: 3)
+                .WithStatusPublished()
+                .WithDataSetId(dataSet.Id)
+                .ForIndex(0, dsv => dsv.SetVersionNumber(1, 0))
+                .ForIndex(1, dsv => dsv.SetVersionNumber(1, 1))
+                .ForIndex(2, dsv => dsv.SetVersionNumber(1, 2))
+                .ForIndex(3, dsv => dsv.SetVersionNumber(3, 5))
+                .ForIndex(4, dsv => dsv.SetVersionNumber(3, 4))
+                .ForIndex(5, dsv => dsv.SetVersionNumber(3, 3))
+                .ForIndex(6, dsv => dsv.SetVersionNumber(3, 2))
+                .ForIndex(7, dsv => dsv.SetVersionNumber(3, 1))
+                .ForIndex(8, dsv => dsv.SetVersionNumber(3, 0))
+                .ForIndex(9, dsv => dsv.SetVersionNumber(2, 0))
+                .ForIndex(10, dsv => dsv.SetVersionNumber(2, 1))
+                .ForIndex(11, dsv => dsv.SetVersionNumber(2, 2))
+                .GenerateList();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.AddRange(dataSetVersions);
+            });
+
+            var response = await ListVersions(
+                dataSetId: dataSet.Id,
+                page: 2,
+                pageSize: 5);
+
+            var content = response.AssertOk<DataSetVersionPaginatedListViewModel>(useSystemJson: true);
+
+            Assert.NotNull(content);
+            Assert.Equal(2, content.Paging.Page);
+            Assert.Equal(5, content.Paging.PageSize);
+            Assert.Equal(12, content.Paging.TotalResults);
+            Assert.Equal(5, content.Results.Count);
+
+            Assert.Equal("3.0", content.Results[0].Number);
+            Assert.Equal("2.2", content.Results[1].Number);
+            Assert.Equal("2.1", content.Results[2].Number);
+            Assert.Equal("2.0", content.Results[3].Number);
+            Assert.Equal("1.2", content.Results[4].Number);
```

Comment:
Given we're generating multiple pages of versions, we might want to just check it orders correctly across all the pages as well. We could do this as successive requests for each page.

However, would consider bringing down the number of versions generated to shorten the test as 12 items is a bit overkill.

Reply:
The reason I had 12 items before was to make sure that the page I requested had some overlap into each major version. And I wanted to make sure each major version had at least 3 minor versions to make sure that things were being ordered correctly (could have been 2 I guess). Then I was just testing the middle page.

But I guess it would be more thorough to test each page. And if we do, then there is less need for 12 items. So I'll change it :)

Reply:
I'm going to still make enough items so that some pages have major version overlaps though as I think that's somewhat nice to test

Reply:
I've also removed the checks for the page properties as we're already testing this in the other pagination tests

Reply:
Nice one, looks good! :+1: